### PR TITLE
[PREVIEW] - ROC-4049: Allow claimant suggest payment plan

### PR DESCRIPTION
**Pull request by geek-soft**
### JIRA link

ROC-4049

### Change description

Changes that allows claimant suggest payment plan when they rejected one proposed by defendant.

Still outstanding:
- update check and send page
- retrofit few small pieces of logic removed / commented out

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/common/components/payment-intention/payment-plan.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    return createPaymentPlan(totalAmount, instalmentAmount, mapFrequencyInWeeks(paymentSchedule)).getPaymentLength()
+  }
+
+  return undefined
+}
+
+export abstract class AbstractPaymentPlanPage<Draft> {
+  abstract getHeading (): string
+  abstract createModelAccessor (): AbstractModelAccessor<Draft, PaymentIntention>
+  abstract buildTaskListUri (req: express.Request, res: express.Response): string
```
Comment (timja) [inline src/main/common/components/payment-intention/payment-plan.ts]: would `taskListUri` be a more accurate name for what this is?

File: src/main/common/components/payment-intention/payment-date.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            const user: User = res.locals.user
+            await new DraftService().save(res.locals.draft, user.bearerToken)
+
+            res.redirect(this.buildTaskListUri(req, res))
+          }
+        }))
+  }
+
+  private renderView (form: Form<PaymentDate>, res: express.Response) {
+    const futureDate: Moment = MomentFactory.currentDate().add(1, 'month')
```
Comment (robertparkinson) [inline src/main/common/components/payment-intention/payment-date.ts]: We need to be consistent with how we do example dates please update to use similar logs as this example: 
https://github.com/hmcts/cmc-citizen-frontend/pull/804/files#diff-f513f2f23d4687729ac795343d516a45R84

File: src/main/features/claimant-response/views/payment-option.njk [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{% extends "layout.njk" %}
+
+{% from "form.njk" import saveAndContinueButton, csrfProtection, radioGroup, radioGroupWithHintPanel, errorSummary %}
```
[the remark this answers is not in the document]
Comment (geek-soft) [reply src/main/features/claimant-response/views/payment-option.njk]: Done. Thanks @sabahirfan

File: src/main/features/claimant-response/views/payment-plan.njk [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      <h2 class="heading-small">{{ t('Their finances') }}</h2>
+      <p>{{ t('Their total monthly income is {{ monthlyIncomeAmount }}',
+          { monthlyIncomeAmount: monthlyIncomeAmount | numeral }) }}</p>
+      <p>{{ t('Their total monthly expenses are {{ monthlyExpensesAmount }}',
+          { monthlyExpensesAmount: monthlyExpensesAmount | numeral }) }}</p>
+      {{ financialDetailsTable(label = 'See their financial details', statementOfMeans = statementOfMeans, classes = 'form-group') }}
+
+      <input type="hidden" name="totalAmount" value="{{ totalAmount }}"/>
+
+      <div class="form-group">
```
[the remark this answers is not in the document]
Comment (geek-soft) [reply src/main/features/claimant-response/views/payment-plan.njk]: To be honest I have no idea why that was there, I copied template from already existing page. Having these extra from groups does not make sense to me. I removed these.

File: src/main/features/claimant-response/views/payment-plan.njk [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          </fieldset>
+        </div>
+      </div>
+
+      {{ linkButton('Calculate length of repayment plan', 'action[calculatePaymentPlan]', 'button button-secondary calculateLengthOfRepayment') }}
+
+      <h2 class="heading-small">{{ t('Length of repayment plan') }}</h2>
+      <p id="lengthOfRepayment" class="lengthOfRepayment"> {{ paymentLength | default('-') }} </p>
+
+      <div class="form-group">
```
[the remark this answers is not in the document]
Comment (geek-soft) [reply src/main/features/claimant-response/views/payment-plan.njk]: The same here.

File: src/main/common/components/payment-intention/payment-date.ts (hunk 2 above)
Comment (geek-soft) [reply src/main/common/components/payment-intention/payment-date.ts]: Good spot, reimplemented to use Andrew's approach.

**Pull request merged**